rest_api/src/main/java/tqs/cloudit/services/AuthenticationService.java

```diff
@@ -24,30 +23,49 @@
  */
 @Service
 public class AuthenticationService {
+    
     @Autowired
     private UserRepository userRepository;
     
     @Autowired
     private AreaRepository areaRepository;
     
-    private BCryptPasswordEncoder bcpe=new BCryptPasswordEncoder();
+    private BCryptPasswordEncoder bcpe = new BCryptPasswordEncoder();
     private Logger l = LoggerFactory.getLogger(AuthenticationService.class);
     
     public ResponseEntity register(User user) {
+        if(!user.allDefined()){
+            JSONObject response = new JSONObject();
+            response.put("status", 1);
+            response.put("message", "Registration invalid. When registering you need to provide username, password, name, email, type of user and optionally interest areas.");
+            return ResponseEntity.badRequest().body(response);
+        }
+        if(userRepository.getUsernames().contains(user.getUsername())){
```

Comment:
I think you should leave this, checking if the username already exists in the user set, to SQL
```SQL
        SELECT count(*) FROM user WHERE username = ?1;
```
```suggestion
        if(userRepository.usernameExists() > 0){
```
Same applies to email